warehouse/slave.go

```diff
@@ -248,7 +246,10 @@ func processStagingFile(job PayloadT) (loadFileIDs []int64, err error) {
 		job:          job,
 		whIdentifier: warehouseutils.GetWarehouseIdentifier(job.DestinationType, job.SourceID, job.DestinationID),
 	}
-	defer jobRun.cleanup()
+	allLoadFilesCleaned := false
```

Comment:
After our prior discussion, I still feel this a bit hacky. Let's delete the files only in cleanup() and not inside the for loop in this function. WDYT?